sql,gossip,systemconfigwatcher: miscellaneous cleanup around zone configs

yuzefovich opened the pull request:
See individual commits for details.

Epic: None

Release note: None

yuzefovich committed:
sql: simplify accumulation of constraints a bit

Also adjust a comment a bit.

Release note: None

yuzefovich committed:
gossip: remove no longer used system config

Note that we cannot fully remove handling of `KeyDeprecatedSystemConfig`
because it requires a migration.

Release note: None

yuzefovich committed:
systemconfigwatcher: move TestSystemConfigWatcher

Previously, due to the requirement for having CCL license to start
tenants we separated out `TestSystemConfigWatcher` into two, but that is
no longer needed.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110756)
<!-- Reviewable:end -->

yuzefovich committed:
gossip: remove unused default zone config reference

Release note: None

yuzefovich committed:
systemconfigwatcher: remove no longer used timestamp in Cache

Previously this was used in tests but they have been deleted at some
point.

Release note: None

yuzefovich commented:
Friendly ping @knz @stevendanna. Perhaps @andrewbaptist you could take a look too.

knz reviewed (approved):
nice. :100: 

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 7 of 7 files at r3, 25 of 25 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110756#-:-NfC96CK3zlc-RuJvj8D:bu517g1)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stevendanna)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
